src/QMCHamiltonians/SOECPComponent.cpp

```diff
@@ -193,6 +190,93 @@ SOECPComponent::RealType SOECPComponent::evaluateOne(ParticleSet& W,
   return std::real(pairpot);
 }
 
+SOECPComponent::RealType SOECPComponent::evaluateValueAndDerivatives(ParticleSet& W,
+                                                                     int iat,
+                                                                     TrialWaveFunction& Psi,
+                                                                     int iel,
+                                                                     RealType r,
+                                                                     const PosType& dr,
+                                                                     const opt_variables_type& optvars,
+                                                                     const Vector<ValueType>& dlogpsi,
+                                                                     Vector<ValueType>& dhpsioverpsi)
+{
+#ifndef QMC_COMPLEX
+  throw std::runtime_error("SOECPComponent::evaluateValueAndDerivatives should not be called in real build\n");
+#else
+  if (sknot_ < 2)
+    APP_ABORT("Spin knots must be greater than 2\n");
```

Comment:
Not "greater equal"?
Could you move this check where sknot_ is set?
APP_ABORT better using exception.

Reply:
fixed